### ISSUE TITLE
InterstitialScreen: accessibility testing issue set 2

### ISSUE THREAD

**Issue by amal-k-joy** [carbon-design-system/ibm-products#7460]
These are the set 2 of issues raised by Raghu on his accessibility testing.

[Carbon for IBM Components_Interstitialscreen_12thMay_Updated.xlsx](https://github.com/user-attachments/files/20186087/Carbon.for.IBM.Components_Interstitialscreen_12thMay_Updated.xlsx)

Issue   title/Summary | Issue Description | Recommended Fix | User Impact | Developer comments
-- | -- | -- | -- | --
After closing the  modal,   focus is not returning to the triggered element | After closing   the  modal, focus is not returning to   the triggered element | After closing   the  modal, focus  expected to set back to the  triggered element | After a user   action, if the focus randomly jumps to another spot on the web page, it’s   tedious to navigate back to the relevant element. This negatively affects the   experience of keyboard-only users and those with cognitive disabilities. | Introduced a new   prop `launcherButtonRef` to achieve this similar to components
Screen reader not announced the status  for the    Step indicators -  where  the status of each steps  indicated visually. | Status   indicators are presented as Buttons  -   here  focusing on each Step indicators   Screen reader not announced the status    -  where  the status of each steps  indicated visually. The announcing status   is not clearer to the screen reader user. | Make   the status   indication  available for screen reader users as well   :     Focusing on each  Stepper Elements  screen reader expected to announce  its    status .     For example : Step 1  completed        Step 2  IN progress /Current      Step 3  Not completed  so on | When   the name, role or state of an element is not conveyed appropriately, it   prevents screen reader users to interact smoothly with the page element. | Step   change is announced to user using `aria-live`
Stepper Buttons are not keyboard /Mouse operable . | Stepper   indicators are presented as Button for screen reader users  , and visually presented as Texts :      here - these elements are not Keyboard    or Mouse operable - unable to    activate the  Steps by these   controls .     Please Confirm  the functional  purpose of these controls ? | 1.   If the Stepper controls   expected to   act interactive controls :     Make the control reachable /operable    via  Standard  keyboard controls  and Mouse      2. If these are   not expected to be   Interactive in nature :     Remove the Tab focus  as well as Role   Button  from these  steppers    to avoid  confusion  for    Screen reader /Keyboard users. | When   the standard keyboard navigation method is not supported, it impacts the   user’s ability to interact with page controls. This defect affects screen   reader and keyboard-only users. | Expected   behaviour for carbon progress indicator which is not keyboard operable
